apps/emqx/src/emqx_session.erl

```diff
@@ -808,13 +799,43 @@ replay(ClientInfo, Session = #session{inflight = Inflight}) ->
     end.
 
 -spec terminate(emqx_types:clientinfo(), Reason :: term(), session()) -> ok.
-terminate(ClientInfo, discarded, Session) ->
-    run_hook('session.discarded', [ClientInfo, info(Session)]);
-terminate(ClientInfo, takenover, Session) ->
-    run_hook('session.takenover', [ClientInfo, info(Session)]);
 terminate(ClientInfo, Reason, Session) ->
+    run_terminate_hooks(ClientInfo, Reason, Session),
+    redispatch_shared_messages(Session),
+    ok.
+
+run_terminate_hooks(ClientInfo, discarded, Session) ->
+    run_hook('session.discarded', [ClientInfo, info(Session)]);
+run_terminate_hooks(ClientInfo, takeovered, Session) ->
+    run_hook('session.takeovered', [ClientInfo, info(Session)]);
```

Comment:
```suggestion
run_terminate_hooks(ClientInfo, takenover, Session) ->
    run_hook('session.takenover', [ClientInfo, info(Session)]);
```

Reply:
While this is a reasonable change, it may affect 3rd party plugins?

Reply:
The previous code already uses `takenover`:

https://github.com/emqx/emqx/pull/7860/files#diff-1dcb97f8eab360778099824b93b350440db45d1d601286f97ef6d4487711f475L813

Reply:
Nice, thanks for the answer!